hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/domain/ContractResultServiceImpl.java

```diff
@@ -82,7 +85,31 @@ public void process(@NonNull RecordItem recordItem, Transaction transaction) {
         // in pre-compile case transaction is not a contract type and entityId will be of a different type
         var contractId = isContractCreateOrCall(recordItem.getTransactionBody()) ? transaction.getEntityId() :
                 entityIdService.lookup(functionResult.getContractID());
-        processContractResult(recordItem, contractId, functionResult, transactionHandler);
+        var contractResult = processContractResult(recordItem, contractId, functionResult, transactionHandler);
+
+        var sidecarRecords = recordItem.getSidecarRecords();
+        var contractActions = new ArrayList<com.hedera.services.stream.proto.ContractAction>();
+        var contractStateChanges = new ArrayList<ContractStateChange>();
+        for (int i = 0; i < sidecarRecords.size(); i++) {
+            var sidecarRecord = sidecarRecords.get(i);
+            if (sidecarRecord.hasStateChanges() && contractResult != null) {
+                var stateChanges = sidecarRecord.getStateChanges();
+                for (int j = 0; j < stateChanges.getContractStateChangesCount(); j++) {
+                    contractStateChanges.add(stateChanges.getContractStateChanges(j));
+                }
+            } else if (sidecarRecord.hasActions()) {
+                var actions = sidecarRecord.getActions();
+                for (int j = 0; j < actions.getContractActionsCount(); j++) {
+                    contractActions.add(actions.getContractActions(j));
+                }
+            }
+        }
+
+        if (contractResult != null) {
+            processContractStateChanges(contractStateChanges, contractResult);
+        }
+
```

Comment:
For migrated state changes, they won't have an associated contract result. Also, not sure if it's possible to get state changes and no contract result. I believe it's not possible, but either way we should actually change state changes to get info from `RecordItem` and not `ContractFunctionResult`. We don't need the special `lookup(contractResult.getContractId(), stateChange.getContractId())` logic and can just use `stateChange.getContractId()`.
```suggestion
        processContractStateChanges(contractStateChanges, recordItem);
```

Reply:
Awesome, with that input I have removed the lookup and no longer use the `ContractFunctionResult`.